augur/parse.py

```diff
@@ -138,6 +138,7 @@ def parse_sequence(
     metadata = dict(zip(fields, sequence_fields))
 
     tmp_name = metadata[strain_key].translate(forbidden_characters)
+    metadata[strain_key] = tmp_name
     sequence.name = sequence.id = tmp_name
```

Comment:
_(non-blocking)_

`tmp_name`'s name is misleading since it's not temporary. It can be removed entirely:

```suggestion
    metadata[strain_key] = metadata[strain_key].translate(forbidden_characters)
    sequence.name = sequence.id = metadata[strain_key]
```